docs/Architecture/LargeScale.md

```diff
@@ -21,20 +21,47 @@ layout: doc_page
+
+* **Serialization Version:** (*SerVer*) This is a small integer placed in the preamble of the serialized byte array that indicates the version of the serialized structure for the sketch. This is very similar to Java's [*Class File Format Version*](https://en.wikipedia.org/wiki/Java_class_file). A single *SerVer* may represent multiple structures all based on the same sketch when stored in different states, e.g., *Single Item*, *Compact*, *Updatable*, etc). This *SerVer* changes very rarely, if at all. Of all of our sketches, only a few, e.g., Theta, KLL and Sampling, have had more than one *SerVer* over time. There are and will be many *Software Versions* of the same sketch that still use the same *SerVer*. When we have to update the *SerVer*, we provide the capability in the *Software Version* of the code associated with the new *SerVer* the ability to read and convert the old *SerVer* to the new *SerVer*. This is why our newest *Software Versions* can still read and interpret older *SerVer* serialized sketches that go back to when our project was started at Yahoo (2012), and before we went open-source (2015). Technically speaking this can be described as *Backward-Transient* compatibility [Schema Evolution and Compatibility](https://docs.confluent.io/platform/current/schema-registry/fundamentals/schema-evolution.html) and [Schema Evolution](https://en.wikipedia.org/wiki/Schema_evolution).
+
+From the user's perspective, as long as the *SerVer* is the same, older *Software Versions* should be able to read sketch images created by newer *Software Versions*. But the APIs may be different, obviously. An older *Software Version* will not be able to take advantage of new features introduced in new *Software Versions*, but it should be able to do what it did before. In other words, there will be no loss of access to the serialized sketch and the older *Software Version* capabilities. A user should not need to access the *SerVer*, nonetheless it is always stored in index one of the serialized image. If a sketch is presented with a *SerVer* that it is not compatible with, the sketch should throw an exception and say what the problem is, just like Java does with its *Class File Format Versions*.
+
+#### The Serialized Image of a Sketch
+* The structure (or image) of a serialized sketch is independent of the language from which it was created. 
+* The sketch image only contains little-endian primitives, such as int64, int32, int16, int8, double-64, float-32, UTF-8 strings, and simple array structures of those, which can be easily interpreted in many languages on modern CPUs. We do not support big-endian serialization.
+* The sketch image is unique for each type of sketch.
+* Simply speaking, a sketch image can be viewed as a blob of bytes, which is easily stored and easily transported using many different protocols, including Protobuf, Avro, Thrift, Byte64, etc.
+
```

Comment:
Should we clarify that for some sketches (FrequencyItemSketch iirc) the serialised form between language may not be strictly equal but still be logically equivalent?

Reply:
Excellent point.

Reply:
Fixed. Thank you!